### PR TITLE
Add user authentication routes

### PR DESCRIPTION
**Pull request by yliu0715**
Refactored code to use this structure: https://riptutorial.com/node-js/example/32332/model-routes-controllers-services-code-structure

### CODE REVIEW

**Comment by yliu0715** [comment]
yes

File: src/web/server/app/utils/queries.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Query names take the following form: <ACTION>_<TABLE NAME>_<COLUMNS YOU WANT
```
Comment (Akasora39) [inline src/web/server/app/utils/queries.ts]: You should include some form of documentation that explains why you have two identical tables.

File: src/web/server/app/index.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+// TODO: Set up the app so that cors is only used for development
+app.use(cors());
+
+app.use(express.json());
+app.use(express.urlencoded({ extended: true }));
+
+app.use(
+  session({
+    secret: process.env.SESSION_SECRET || 'likeasomebooody',
```
Comment (Akasora39) [inline src/web/server/app/index.ts]: Bruh

File: src/web/server/app/middlewares/authChecks.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  req: express.Request,
+  res: express.Response,
+  next: express.NextFunction
+) {
+  const { username, password } = req.body;
+  if (username && password) {
+    return next();
+  } else {
+    res.status(400).send('Bad username or password');
+  }
```
[the remark this answers is not in the document]
Comment (yliu0715) [reply src/web/server/app/middlewares/authChecks.ts]: True, I'll keep this in mind

File: src/web/server/app/index.ts (hunk 2 above)
Comment (yliu0715) [reply src/web/server/app/index.ts]: Yes

File: src/web/server/app/utils/queries.ts (hunk 1 above)
Comment (yliu0715) [reply src/web/server/app/utils/queries.ts]: I can but it's probably not helpful to put it in the `queries.ts` file, it mainly just comes down to the way Cassandra stores information.

**Comment by yliu0715** [comment]
@Akasora39 The email field can be added on the `create_db.py` file, but it is true that I will need to check for existing emails in the database here. I can just do that later, because this PR is already as big enough as it is.

**Comment by yliu0715** [comment]
Yeah thats right bro

**Pull request merged**